pilot/pkg/model/push_context.go

```diff
@@ -759,6 +767,13 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 	return nil
 }
 
+// IsClusterLocal indicates whether the endpoints for the service should only be accessible to clients
+// within the cluster.
+func (ps *PushContext) IsClusterLocal(service *Service) bool {
+	_, ok := MostSpecificHostMatch(service.Hostname, ps.clusterLocalHosts)
```

Comment:
this is very expensive to search through the list. May be an optimization would be to have the cluster local namespaces as a separate field. you can always look at the service's attributes to determine its namespace, so that you dont have to hardcode a *.svc.cluster.local as a suffix.

Reply:
The API is host based, however, not namespace based. I *think* we can assume that this list is fairly small, so I'm not sure if performance will be noticeably impacted.

Reply:
Yes, can cache ClusterLocalNamespaces as well, so that can get rid of the use of `MostSpecificHostMatch`.

Reply:
I'm not sure I follow the comments in this thread. The values provided in mesh config are hosts, not namespaces. I'm not sure how any of the suggestions here apply in that case ... am I missing something?